test/mocha/integ/rpc/ProviderHealthStateDynamoDbRepository.test.ts

```diff
@@ -0,0 +1,60 @@
+import { setupTables } from '../../dbSetup'
+import { DynamoDBTableProps } from '../../../../bin/stacks/routing-database-stack'
+import { default as bunyan } from 'bunyan'
+import { expect } from 'chai'
+import { ProviderHealthStateRepository } from '../../../../lib/rpc/ProviderHealthStateRepository'
+import { ProviderHealthStateDynamoDbRepository } from '../../../../lib/rpc/ProviderHealthStateDynamoDbRepository'
+import { ProviderHealthState } from '../../../../lib/rpc/ProviderHealthState'
+
+const DB_TABLE = {
+  TableName: DynamoDBTableProps.RpcProviderHealthStateDbTable.Name,
+  KeySchema: [
+    {
+      AttributeName: 'chainIdProviderName',
+      KeyType: 'HASH',
+    },
+  ],
+  AttributeDefinitions: [
+    {
+      AttributeName: 'chainIdProviderName',
+      AttributeType: 'S',
+    },
+  ],
+  ProvisionedThroughput: {
+    ReadCapacityUnits: 1,
+    WriteCapacityUnits: 1,
+  },
+}
+
+const PROVIDER_ID = 'provider_id'
+
+const log = bunyan.createLogger({
+  name: 'ProviderHealthStateDynamoDbRepositoryTest',
+  serializers: bunyan.stdSerializers,
+  level: bunyan.DEBUG,
+})
+
+describe('ProviderHealthStateDynamoDbRepository', () => {
+  setupTables(DB_TABLE)
+  const storage: ProviderHealthStateRepository = new ProviderHealthStateDynamoDbRepository(
+    DynamoDBTableProps.RpcProviderHealthStateDbTable.Name,
+    log
+  )
+
+  it('write state to DB then read from it', async () => {
```

Comment:
this test does not test the functionality of an actual DDB, i suggest moving it to a unit test with a mocked DDB interface

Reply:
This test does use a real DynamoDB instance. It has to be run after starting a local DynamoDB instance. See `setupTables()` function where we configure DDB IP and port number.

Reply:
it's still an actual DDB, even if it is local. It makes the test unecessarily heavy IMHO.

Reply:
Agreed, it's heavier than unit test...
Currently, using `dynamodb-local` is a universally accepted solution in `routing-api`, `api`, `data-api-graphql` and other repos. Fortunately we don't have big payload in the test so they run pretty fast so far.